### PR TITLE
Move template replacement to the frontend

### PR DESCRIPTION
**Pull request by 121watts**
Connect #3767

### CODE REVIEW

File: ui/src/shared/components/DatabaseList.tsx [hunk 1]
```diff
@@ -52,18 +52,13 @@ class DatabaseList extends Component<DatabaseListProps, DatabaseListState> {
...
   }: {
     querySource?: Source
     query: QueryConfig
   }) {
     const {querySource: nextSource, query: nextQuery} = this.props
     const differentSource = !_.isEqual(prevSource, nextSource)
 
-    if (prevQuery.rawText === nextQuery.rawText) {
-      return
-    }
```
Comment (chnn) [inline ui/src/shared/components/DatabaseList.tsx]: how come this was removed?

File: ui/src/tempVars/utils/replace.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  query: string,
+  pixels: number,
+  durationMs: number
+) => {
+  if (!query.includes(TEMP_VAR_INTERVAL)) {
+    return query
+  }
+
+  if (!pixels) {
+    pixels = 333
```
Comment (chnn) [inline ui/src/tempVars/utils/replace.ts]: yea definitely feel we should extract these constants

File: ui/src/tempVars/utils/replace.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  if (!durationMs) {
+    durationMs = 1000
```
Comment (chnn) [inline ui/src/tempVars/utils/replace.ts]: ^

File: ui/src/tempVars/utils/replace.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+const templateReplace = (query: string, tempVars: Template[]) => {
+  const replacedQuery = tempVars.reduce((acc, template) => {
+    return renderTemplate(acc, template)
+  }, query)
+
+  return replacedQuery
+}
+
+const renderTemplate = (query: string, template: Template): string => {
```
Comment (chnn) [inline ui/src/tempVars/utils/replace.ts]: super clean, nice

**Comment by chnn** [comment]
Also found maybe a bug?

Check this out:

kapture 2018-06-28 at 13 17 16

If I enter a template variable into a query that makes an invalid request, the `:interval:` is replaced by the actual duration once the error returns. I think it might make more sense if the `:interval:` stayed in place.

File: ui/src/shared/components/DatabaseList.tsx (hunk 1 above)
Comment (121watts) [reply ui/src/shared/components/DatabaseList.tsx]: i actually do want the query to run again if I change sources and the query is still the same.

File: ui/src/tempVars/utils/replace.ts (hunk 2 above)
Comment (121watts) [reply ui/src/tempVars/utils/replace.ts]: consider it done.  we used to have a linter rule for magic numbers and it was removed.  mayhaps we add it back?

**Comment by 121watts** [comment]
Fixed the bug 😸

**Pull request merged**